src/internal/components/chart-legend/index.tsx

```diff
@@ -120,44 +126,49 @@ function ChartLegend<T>({
           </InternalBox>
         )}
 
-        <ul className={styles.list}>
+        <div className={styles.list}>
           {series.map((s, index) => {
             const someHighlighted = highlightedSeries !== null;
             const isHighlighted = highlightedSeries === s.datum;
             const isDimmed = someHighlighted && !isHighlighted;
             return (
-              <li
+              <div
+                {...focusVisible}
+                role="button"
                 key={index}
-                onMouseOver={handleMouseOver.bind(null, s.datum)}
-                onMouseLeave={handleMouseLeave}
                 className={clsx(styles.marker, {
                   [styles['marker--dimmed']]: isDimmed,
                   [styles['marker--highlighted']]: isHighlighted,
-                  [styles['marker--focused']]: isHighlighted && isFocused,
                 })}
-                aria-disabled={isDimmed ? true : undefined}
+                ref={elem => {
+                  if (elem) {
+                    segmentsRef.current[index] = elem;
+                  } else {
+                    delete segmentsRef.current[index];
+                  }
+                }}
+                tabIndex={
+                  index === highlightedSeriesIndex || (highlightedSeriesIndex === undefined && index === focusedIndex)
```

Comment:
Why the second check is needed? Can the legend be focused but not highlighted at the same time?